src/VisualStudio/Core/Def/Implementation/Diagnostics/DiagnosticProgressReporter.cs

```diff
@@ -18,192 +18,181 @@ internal sealed class TaskCenterSolutionAnalysisProgressReporter
+        /// So we have to store the count separately for the UI so that we do not overwrite the last reported count with 0.
+        /// Protected from concurrent access due to serialization from <see cref="_updateUITask"/>
+        /// </summary>
         private int _lastPendingItemCount;
-        private ProgressStatus _lastProgressStatus;
-        private ProgressStatus _lastShownProgressStatus;
-
-        // _resettableDelay makes sure that we at the end update status to correct value.
-        // in contrast to _lastTimeReported makes sure that we only update at least s_minimumInterval interval.
-        //
-        // for example, when an event stream comes in as below (assuming 200ms minimum interval)
-        // e1 -> (100ms)-> e2 -> (300ms)-> e3 -> (100ms) -> e4
-        //
-        // actual status shown to users without _resettableDelay will be 
-        // e1 -> e3.
-        //
-        // e2 and e4 will be skipped since interval was smaller than min interval.
-        // losing e2 is fine, but e4 is problematic since the user now could see the wrong status 
-        // until the next event comes in. 
-        // for example, it could show "Evaluating" when it is actually "Paused" until the next event
-        // which could be long time later.
-        // what _resettableDelay does is making sure that if the next event doesn't come in 
-        // within certain delay, it updates status to e4 (current).
-        private ResettableDelay _resettableDelay;
-
-        // this is only field that is shared between 2 events streams (IDiagnosticService and ISolutionCrawlerProgressReporter)
-        // and can be called concurrently.
-        private volatile ITaskHandler _taskHandler;
+
+        #endregion
 
         [ImportingConstructor]
         public TaskCenterSolutionAnalysisProgressReporter(
             SVsTaskStatusCenterService taskStatusCenterService,
             IDiagnosticService diagnosticService,
             VisualStudioWorkspace workspace)
         {
-            _lastTimeReported = DateTimeOffset.UtcNow;
-            _resettableDelay = null;
-
-            ResetProgressStatus();
-
             _taskCenterService = (IVsTaskStatusCenterService)taskStatusCenterService;
-
             _options = new TaskHandlerOptions()
             {
                 Title = ServicesVSResources.Running_low_priority_background_processes,
                 ActionsAfterCompletion = CompletionActions.None
             };
 
-            var crawlerService = workspace.Services.GetService<ISolutionCrawlerService>();
+            var crawlerService = workspace.Services.GetRequiredService<ISolutionCrawlerService>();
             var reporter = crawlerService.GetProgressReporter(workspace);
 
-            StartedOrStopped(reporter.InProgress);
+            OnSolutionCrawlerProgressChanged(this, new ProgressData(reporter.InProgress ? ProgressStatus.Started : ProgressStatus.Stopped,
+                pendingItemCount: null));
```

Comment:
1. we really only need to issue this in the "InProgress" state. 
2. possibly worth a comment.

(very minor nits).